pkg/resource/deploy/plan.go

```diff
@@ -35,6 +35,10 @@ import (
 type BackendClient interface {
 	// GetStackOutputs returns the outputs (if any) for the named stack or an error if the stack cannot be found.
 	GetStackOutputs(ctx context.Context, name string) (resource.PropertyMap, error)
+
+	// GetStackResourceOutputs returns the resource outputs (if any) for the named stack or an
+	// error if the stack cannot be found.
+	GetStackResourceOutputs(ctx context.Context, stackName, typ string) (resource.PropertyMap, error)
```

Comment:
It isn't immediately obvious what the `typ` parameter does. From the looks of it, it is an optional filter, and if non-empty `GetStackResourceOutputs` will only return the stack outputs with that type? If so, we should probably mention it -- even if as a pithy "takes an optional type filter" -- in the comments.